stripe/src/main/java/com/stripe/android/FingerprintDataRepository.kt

```diff
@@ -20,25 +21,28 @@ internal interface FingerprintDataRepository {
         private val fingerprintRequestFactory: FingerprintRequestFactory,
         private val fingerprintRequestExecutor: FingerprintRequestExecutor =
             FingerprintRequestExecutor.Default(),
-        private val coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.Main + SupervisorJob())
+        dispatcher: CoroutineDispatcher = Dispatchers.IO
     ) : FingerprintDataRepository {
         private var cachedFingerprintData: FingerprintData? = null
 
         private val timestampSupplier: () -> Long = {
             Calendar.getInstance().timeInMillis
         }
 
+        private val scope = CoroutineScope(dispatcher)
+
         constructor(
             context: Context
         ) : this(
             store = FingerprintDataStore.Default(context),
             fingerprintRequestFactory = FingerprintRequestFactory(context)
         )
 
+        @OptIn(FlowPreview::class)
         override fun refresh() {
             if (Stripe.advancedFraudSignalsEnabled) {
-                coroutineScope.launch {
-                    store.get().switchMap { localFingerprintData ->
+                scope.launch {
+                    store.get().flatMapMerge { localFingerprintData ->
```

Comment:
If `get()` switches to `async`/`await`, this can look more "imperative" instead of having to work in the context of the `Flow` container.